engine/src/main/java/com/redhat/cloud/notifications/processors/camel/CamelTypeProcessor.java

```diff
@@ -93,16 +90,20 @@ public List<NotificationHistory> process(Event event, List<Endpoint> endpoints)
     }
 
     private NotificationHistory process(Notification item) {
-        processedCount.increment();
         Endpoint endpoint = item.getEndpoint();
+        String subType = endpoint.getSubType();
+
+        Counter processedCount = registry.counter(PROCESSED_COUNTER_NAME, "subType", subType);
```

Comment:
I'm confused. Do dynamic tags work with Micrometer's counter or not? :smiley: Last time I heard about that, it didn't work.

Reply:
Maybe that was fixed upstream and it's finally working because we bumped Quarkus?

Reply:
They are probably not really dynamic. `registry.counter()` either registers a new counter or gives one back that has the passed name and tags. Meaning I am not using a dynamic tag at the time of `counter.increment()` but create a new counter for each of the passed tags (and re-use later).

Reply:
I *think* it works when the label is added that way, but not if the tags are declared after.
something.tag("blah", "blih") --> didn't work

The question is if `registry.counter(PROCESSED_COUNTER_NAME, "subType", subType)` is a costly operation or not, it used to be declared once.

Reply:
Internally there is a map <Id, Meter>  where Id = <name, Tags>  so that may be more costly on the 1st event, but cheap afterwards.